web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1129,34 +1134,63 @@ namespace com.keyman.keyboards {
         }
       } else if(options['context'] == 'language') { // Download the full list of supported keyboard languages
         this.languageList = x['languages'];
-        if(this.languagesPending) {
-          this.addLanguageKeyboards(this.languagesPending);
-        }
-        this.languagesPending = [];
       }
 
-      return null; // TODO (#5044): return the list of newly-registered stubs here.
+      return this.keyboardStubs.slice(currentKeyboardStubsCount);
     }
 
     /**
      *  Internal handler for processing keyboard registration, used only by `register`
      *
-     *  @param  {Object}   languages    Array of language names
+     *  @param  {string[]}   languages    Array of language names
+     *  @returns {Promise<KeyboardStub[]>} Promise of added keyboard stubs
      **/
-    addLanguageKeyboards(languages) {
-      var i, j, lgName, cmd, first, addAll;
+    addLanguageKeyboards(languages: string[]): Promise<KeyboardStub[]> {
+      var i, j, lgName, cmd, addAll, retPromise;
 
       // Defer registering keyboards by language until the language list has been loaded
-      if(this.languageList == null) {
-        first = (this.languagesPending.length == 0);
-
-        for(i=0; i<languages.length; i++) {
-          this.languagesPending.push(languages[i]);
+      if (this.languageList == null) {
+        if (this.firstCall) {
+          this.firstCall = false;
+          let promise = this.keymanCloudRequest('',true);
+          // If promise is not error, then... (needs an error guard)
+          promise.catch(function(error) {
+            console.log('first error', error);
```

Comment:
This looks like an in-development test message, rather than an intended-for-production log message.

If we want to keep a log message here, it should probably be a `console.error` with a more informative message.  Something like:

```suggestion
            console.error("Unable to retrieve the master language list.")
```

That would likely occur if internet connectivity was poor, so we could also likely reset `this.firstCall` (since it failed) to allow a retry for future calls.